core/src/host_imports/lowering/lo_helper.rs

```diff
@@ -15,34 +15,57 @@
  */
 
 use super::AllocateFunc;
-use crate::module::wit_prelude::WITMemoryView;
-
 use crate::call_wasm_func;
+
 use it_lilo::traits::Allocatable;
 use it_lilo::traits::AllocatableError;
+use it_memory_traits::MemoryView;
+use it_memory_traits::Memory;
+use marine_wasm_backend_traits::{DelayedContextLifetime, WasmBackend};
 
-use it_lilo::traits::DEFAULT_MEMORY_INDEX;
-use wasmer_core::vm::Ctx;
+use std::marker::PhantomData;
 
-pub(crate) struct LoHelper<'c> {
-    allocate_func: &'c AllocateFunc,
-    ctx: &'c Ctx,
+pub(crate) struct LoHelper<
+    'c,
+    WB: WasmBackend,
+    MV: MemoryView<DelayedContextLifetime<WB>>,
```

Comment:
`Memory` trait from IT requires `MemoryView` parameter, so i have to provide it, and thus name it, and also i have to bound it with `MemoryView` trait to be able to pass it to `Memory`. We could rewrite IT API without `MemoryView` trait at all, as it is an implementation detail from Wasmer (both old and new). Or make `MemoryView` an associated type instead of type parameter in IT interface.